File: Project_Sourcecode_go/abstractfactory/AbstractFactory.go
package abstractfactory

import (
	"cs586_project/output/strategy/displayBalance"
	"cs586_project/output/strategy/displayMenu"
	"cs586_project/output/strategy/incorrectIdMsg"
	"cs586_project/output/strategy/incorrectLockMsg"
	"cs586_project/output/strategy/incorrectPinMsg"
	"cs586_project/output/strategy/incorrectUnlockMsg"
	"cs586_project/output/strategy/makeDeposit"
	"cs586_project/output/strategy/makeWithdraw"
	"cs586_project/output/strategy/noFundsMsg"
	"cs586_project/output/strategy/penalty"
	"cs586_project/output/strategy/promptForPin"
	"cs586_project/output/strategy/storeData"
	"cs586_project/output/strategy/tooManyAttemptsMsg"
)

type AbstractFactory struct {
	Ds                 interface{}
	displayBalance     displayBalance.DisplayBalance
	displayMenu        displayMenu.DisplayMenu
	incorrectIdMsg     incorrectIdMsg.IncorrectIdMsg
	incorrectLockMsg   incorrectLockMsg.IncorrectLockMsg
	incorrectPinMsg    incorrectPinMsg.IncorrectPinMsg
	incorrectUnlockMsg incorrectUnlockMsg.IncorrectUnlockMsg
	makeDeposit        makeDeposit.MakeDeposit
	makeWithdraw       makeWithdraw.MakeWithdraw
	noFundsMsg         noFundsMsg.NoFundsMsg
	penalty            penalty.Penalty
	promptForPin       promptForPin.PromptForPin
	storeData          storeData.StoreData
	tooManyAttemptsMsg tooManyAttemptsMsg.TooManyAttemptsMsg
}

type AbstractFactoryApi interface {
	GetDs() interface{}
	GetDisplayMenu() displayMenu.DisplayMenu
	GetIncorrectIdMsg() incorrectIdMsg.IncorrectIdMsg
	GetIncorrectLockMsg() incorrectLockMsg.IncorrectLockMsg
	GetIncorrectPinMsg() incorrectPinMsg.IncorrectPinMsg
	GetIncorrectUnlockMsg() incorrectUnlockMsg.IncorrectUnlockMsg
	GetMakeDeposit() makeDeposit.MakeDeposit
	GetMakeWithdraw() makeWithdraw.MakeWithdraw
	GetNoFundsMsg() noFundsMsg.NoFundsMsg
	GetPenalty() penalty.Penalty
	GetPromptForPin() promptForPin.PromptForPin
	GetStoreData() storeData.StoreData
	GetTooManyAttemptsMsg() tooManyAttemptsMsg.TooManyAttemptsMsg
	GetDisplayBalance() displayBalance.DisplayBalance
}

func (af *AbstractFactory) GetDs() interface{} {
	return nil
}

func (af *AbstractFactory) GetDisplayMenu() displayMenu.DisplayMenu {
	return nil
}

func (af *AbstractFactory) GetIncorrectIdMsg() incorrectIdMsg.IncorrectIdMsg {
	return nil
}

func (af *AbstractFactory) GetIncorrectLockMsg() incorrectLockMsg.IncorrectLockMsg {
	return nil
}

func (af *AbstractFactory) GetIncorrectPinMsg() incorrectPinMsg.IncorrectPinMsg {
	return nil
}

func (af *AbstractFactory) GetIncorrectUnlockMsg() incorrectUnlockMsg.IncorrectUnlockMsg {
	return nil
}

func (af *AbstractFactory) GetMakeDeposit() makeDeposit.MakeDeposit {
	return nil
}

func (af *AbstractFactory) GetMakeWithdraw() makeWithdraw.MakeWithdraw {
	return nil
}

func (af *AbstractFactory) GetNoFundsMsg() noFundsMsg.NoFundsMsg {
	return nil
}

func (af *AbstractFactory) GetPenalty() penalty.Penalty {
	return nil
}

func (af *AbstractFactory) GetPromptForPin() promptForPin.PromptForPin {
	return nil
}

func (af *AbstractFactory) GetStoreData() storeData.StoreData {
	return nil
}

func (af *AbstractFactory) GetTooManyAttemptsMsg() tooManyAttemptsMsg.TooManyAttemptsMsg {
	return nil
}

func (af *AbstractFactory) GetDisplayBalance() displayBalance.DisplayBalance {
	return nil
}

File: Project_Sourcecode_java/src/main/java/edu/iit/cs586/project/abstractfactory/AbstractFactory.java
package edu.iit.cs586.project.abstractfactory;

import edu.iit.cs586.project.datastore.DataStore;
import edu.iit.cs586.project.output.strategy.DisplayBalance.DisplayBalance;
import edu.iit.cs586.project.output.strategy.DisplayMenu.DisplayMenu;
import edu.iit.cs586.project.output.strategy.IncorrectIdMsg.IncorrectIdMsg;
import edu.iit.cs586.project.output.strategy.IncorrectLockMsg.IncorrectLockMsg;
import edu.iit.cs586.project.output.strategy.IncorrectPinMsg.IncorrectPinMsg;
import edu.iit.cs586.project.output.strategy.IncorrectUnlockMsg.IncorrectUnlockMsg;
import edu.iit.cs586.project.output.strategy.MakeDeposit.MakeDeposit;
import edu.iit.cs586.project.output.strategy.MakeWithdraw.MakeWithdraw;
import edu.iit.cs586.project.output.strategy.NoFundsMsg.NoFundsMsg;
import edu.iit.cs586.project.output.strategy.Penalty.Penalty;
import edu.iit.cs586.project.output.strategy.PromptForPin.PromptForPin;
import edu.iit.cs586.project.output.strategy.StoreData.StoreData;
import edu.iit.cs586.project.output.strategy.TooManyAttemptsMsg.TooManyAttemptsMsg;

/**
 * @author CharmingLee <charminglee@jieztech.com>
 * @date 2022/5/1 21:43
 */
public abstract class AbstractFactory {
    protected DataStore ds;

    /**
     * Actions
     */
    // displays the current value of the balance
    protected DisplayBalance displayBalance;
    // display a menu with a list of transactions
    protected DisplayMenu displayMenu;
    // displays incorrect ID message
    protected IncorrectIdMsg incorrectIdMsg;
    // displays incorrect lock msg
    protected IncorrectLockMsg incorrectLockMsg;
    // displays incorrect pin message
    protected IncorrectPinMsg incorrectPinMsg;
    // displays incorrect unlock msg
    protected IncorrectUnlockMsg incorrectUnlockMsg;
    // makes deposit (increases balance by a value stored in temp_d in data store)
    protected MakeDeposit makeDeposit;
    // makes withdraw (decreases balance by a value stored in temp_w data store)
    protected MakeWithdraw makeWithdraw;
    // Displays no sufficient funds msg
    protected NoFundsMsg noFundsMsg;
    // applies penalty (decreases balance by the amount of penalty)
    protected Penalty penalty;
    // prompts to enter pin
    protected PromptForPin promptForPin;
    // stores data from temporary area in data store
    protected StoreData storeData;
    // display too many attempts message
    protected TooManyAttemptsMsg tooManyAttemptsMsg;

    public abstract DataStore getDataStore();

    /**
     * Actions
     */
    public abstract DisplayBalance getDisplayBalance();
    public abstract DisplayMenu getDisplayMenu();
    public abstract IncorrectIdMsg getIncorrectIdMsg();
    public abstract IncorrectLockMsg getIncorrectLockMsg();
    public abstract IncorrectPinMsg getIncorrectPinMsg();
    public abstract IncorrectUnlockMsg getIncorrectUnlockMsg();
    public abstract MakeDeposit getMakeDeposit();
    public abstract MakeWithdraw getMakeWithdraw();
    public abstract NoFundsMsg getNoFundsMsg();
    public abstract Penalty getPenalty();
    public abstract PromptForPin getPromptForPin();
    public abstract StoreData getStoreData();
    public abstract TooManyAttemptsMsg getTooManyAttemptsMsg();
}
